manager/integration/tests/test_recurring_job.py

```diff
@@ -759,3 +794,641 @@ def backup_complete_predicate(b):
+    wait_for_cron_job_create(batch_v1_beta_api, JOB_LABEL+"="+snap1)
+    wait_for_cron_job_create(batch_v1_beta_api, JOB_LABEL+"="+back1)
+
+    volume2_name = "test-policy-2"
+    client.create_volume(name=volume2_name, size=SIZE)
+    volume2 = common.wait_for_volume_detached(client, volume2_name)
+    volume2.attach(hostId=get_self_host_id())
+    volume2 = wait_for_volume_healthy(client, volume2_name)
+    wait_for_volume_recurring_job_update(volume2, jobs=[], groups=[default])
+
+    volume2.recurringJobAdd(name=snap2, isGroup=False)
+    volume2.recurringJobAdd(name=back2, isGroup=False)
+    wait_for_cron_job_count(batch_v1_beta_api, 4)
+    wait_for_cron_job_create(batch_v1_beta_api, JOB_LABEL+"="+snap1)
+    wait_for_cron_job_create(batch_v1_beta_api, JOB_LABEL+"="+snap2)
+    wait_for_cron_job_create(batch_v1_beta_api, JOB_LABEL+"="+back1)
+    wait_for_cron_job_create(batch_v1_beta_api, JOB_LABEL+"="+back2)
+    wait_for_volume_recurring_job_update(volume1,
+                                         jobs=[], groups=[default])
+    wait_for_volume_recurring_job_update(volume2,
+                                         jobs=[snap2, back2], groups=[])
+
+
+@pytest.mark.recurring_job  # NOQA
+def test_recurring_job_policy_snapshot(client, batch_v1_beta_api):  # NOQA
+    """
+    Scenario: test recurring job snapshot
+
+    """
+    volume1_name = "test-policy-1"
+    volume2_name = "test-policy-2"
+    client.create_volume(name=volume1_name, size=SIZE)
+    client.create_volume(name=volume2_name, size=SIZE)
+    volume1 = common.wait_for_volume_detached(client, volume1_name)
+    volume2 = common.wait_for_volume_detached(client, volume2_name)
+
+    self_host = get_self_host_id()
+
+    volume1.attach(hostId=self_host)
+    volume2.attach(hostId=self_host)
+    volume1 = wait_for_volume_healthy(client, volume1_name)
+    volume2 = wait_for_volume_healthy(client, volume2_name)
+
+    snap1 = SNAPSHOT + "1"
+
+    default = "default"
+
+    recurring_jobs = {
+        snap1: {
+            TASK: SNAPSHOT,
+            GROUPS: [default],
+            CRON: "* * * * *",
+            RETAIN: 2,
+            CONCURRENCY: 2,
+        },
+    }
+    create_recurring_jobs(client, recurring_jobs)
+    wait_for_cron_job_count(batch_v1_beta_api, 1)
+
+    # volume-head
+    wait_for_snapshot_count(volume1, 1)
+    wait_for_snapshot_count(volume2, 1)
+
+    # 1st job
+    common.write_volume_random_data(volume1)
+    common.write_volume_random_data(volume2)
+    time.sleep(60)
+    wait_for_snapshot_count(volume1, 2)
+    wait_for_snapshot_count(volume2, 2)
+
+    # 2nd job
+    # wait_until_begin_of_a_minute()
+    common.write_volume_random_data(volume1)
+    common.write_volume_random_data(volume2)
+    time.sleep(60)
+    wait_for_snapshot_count(volume1, 3)
+    wait_for_snapshot_count(volume2, 3)
+
+
+@pytest.mark.recurring_job  # NOQA
+def test_recurring_job_policy_backup(set_random_backupstore, client, batch_v1_beta_api):  # NOQA
```

Comment:
Probably we can remove policy keyword

Reply:
Oops. Thanks, I've missed a few... fixed.